### PR TITLE
Happy thoughts API - Malin Vannesjö

### PR DESCRIPTION
**Pull request by malinvan**
https://malins-happy-thoughts.herokuapp.com

https://malins-happy-thoughts.netlify.app

### CODE REVIEW

**Review by matildasvensson85 — APPROVED** [review]
Nice and clean code, easy to understand and follow. Great job! :)

File: README.md [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
-
+This weeks assignment was to create an API backend to add to the previous Happy Thoughts project.
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
-
+The code was pretty straight forward. My issue was more with the connection to Atlas. I created a cluster for last weeks project and tried hard and long to add this weeks collection. The data was sent but there was some form of issue with the connection to Atlas. I ended up getting help which didn't solve the issue. In the end I had to create a new account and then magically everything worked.
 ## View it live
```
Comment (KarolCislowski) [inline README.md:8]: short documentation would be nice 🤓

File: server.js [hunk 2]
```diff
@@ -1,30 +1,109 @@
...
+});
+mongoose.Promise = Promise;
+mongoose.set("debug", { shell: true });
+
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    minlength: 5,
+    maxlength: 140,
+    required: true,
```
Comment (KarolCislowski) [inline server.js:19]: Nicely configured field.

File: server.js [hunk 3]
```diff
@@ -1,30 +1,109 @@
...
+
+app.get("/", (req, res) => {
+  res.send("Hello from the other side 🎶");
+});
+
+app.get("/thoughts", async (req, res) => {
+  const thoughts = await Thought.find()
+    .sort({ createdAt: "desc" })
+    .limit(20)
+    .exec();
```
Comment (KarolCislowski) [inline server.js:60]: 👍🏻

File: server.js [hunk 4]
```diff
@@ -1,30 +1,109 @@
...
+});
+
+app.post("/thoughts/:id/likes", async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const updatedThought = await Thought.findOneAndUpdate(
+      { _id: id },
+      { $inc: { hearts: 1 } },
+      { new: true }
```
Comment (KarolCislowski) [inline server.js:81]: 💯

File: server.js [hunk 5]
```diff
@@ -1,30 +1,109 @@
...
+  try {
+    const deletedThought = await Thought.findOneAndDelete({ _id: id });
+    if (deletedThought) {
+      res.json(deletedThought);
+    } else {
+      res.status(404).json({ message: "Not found!" });
+    }
+  } catch (error) {
+    res.status(400).json({ message: "Invalid request", error });
+  }
```
Comment (KarolCislowski) [inline server.js:104]: Remember to restrict access to this endpoint because not all users should have access to this functionality 🤓

**Review by KarolCislowski — APPROVED** [review]
Great API 🦄

**Pull request closed without merging**